src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -226,6 +245,96 @@ private void initializeNucleusComponentConfig() {
+                    .filter(s -> !s.startsWith(JVM_OPTION_ROOT_PATH)).collect(Collectors.joining(" "));
+            kernel.getConfig().lookup(SERVICES_NAMESPACE_TOPIC, getNucleusComponentName(), CONFIGURATION_CONFIG_KEY,
+                    DEVICE_PARAM_JVM_OPTIONS).dflt(jvmOptions);
+
+            kernelAlts.writeLaunchParamsToFile(jvmOptions);
+            logger.atInfo().log("Successfully setup Nucleus launch parameters");
+        } catch (IOException e) {
+            logger.atError().log("Unable to setup Nucleus launch parameters", e);
+        }
+    }
+
+    /**
+     * Load Nucleus component information from build recipe.
+     *
+     * @param kernelAlts KernelAlternatives instance
+     */
+    public void initializedNucleusFromRecipe(KernelAlternatives kernelAlts) {
+        String nucleusComponentName = getNucleusComponentName();
+        Topics nucleusLifecycle = kernel.getConfig().lookupTopics(DEFAULT_VALUE_TIMESTAMP, SERVICES_NAMESPACE_TOPIC,
+                nucleusComponentName, SERVICE_LIFECYCLE_NAMESPACE_TOPIC);
+        if (nucleusLifecycle.children.size() != 0) {
+            logger.atDebug().log("Nucleus lifecycle has already been initialized");
+            return;
+        }
+
+        persistInitialLaunchParams(kernelAlts);
+        try {
+            Path unpackDir = locateCurrentKernelUnpackDir();
+            Path recipePath = unpackDir.resolve(NUCLEUS_BUILD_METADATA_DIRECTORY)
+                    .resolve(NUCLEUS_RECIPE_FILENAME);
+
+            // Copy recipe to component store
+            ComponentStore componentStore = kernel.getContext().get(ComponentStore.class);
+            DeploymentService.copyRecipeFileToComponentStore(componentStore, recipePath, logger);
+
+            // Update Nucleus in config store
+            Optional<ComponentRecipe> resolvedRecipe = kernel.getContext().get(RecipeLoader.class)
+                    .loadFromFile(new String(Files.readAllBytes(recipePath.toAbsolutePath()), StandardCharsets.UTF_8));
+            if (!resolvedRecipe.isPresent()) {
+                throw new PackageLoadingException("Failed to resolve Nucleus recipe");
+            }
+            ComponentRecipe componentRecipe = resolvedRecipe.get();
+            nucleusComponentName = componentRecipe.getComponentName();
+
+            // Add Nucleus version
+            Semver componentVersion = componentRecipe.getVersion();
+            kernel.getConfig().lookup(SERVICES_NAMESPACE_TOPIC, nucleusComponentName,
+                    VERSION_CONFIG_KEY).withValue(componentVersion.toString());
+
+            // Copy unpacked artifacts to component store
+            Path unpackArtifactPath = kernel.getContext().get(NucleusPaths.class).unarchiveArtifactPath(
+                    new ComponentIdentifier(nucleusComponentName, componentVersion), "aws.greengrass.nucleus");
+            Utils.copyFolderRecursively(unpackDir, unpackArtifactPath, REPLACE_EXISTING, NOFOLLOW_LINKS,
```

Comment:
this might be copying too much, if I don't unpack into a directory which contains only the zip stuff, like if I unzip into my home directory.

We may want to restrict the copying more. Just something to consider anyway.

Reply:
If it's ok not to copy over the license files, I can explicitly copy the bin/ conf/ lib/ directories only

Reply:
Seems ok to me.
@jbutler @philcali  any issues with this? We still distribute the licenses, we'd just not be copying them into the unzip path.